### ISSUE TITLE
babashka.deps/add-deps support :min-bb-version warning

### ISSUE THREAD

**Issue by waffletower-reify** [babashka/babashka#1468]
Would be great if use of `babashka/add-deps` with a `:min-bb-version` map-entry could log a warning when the babashka version is not sufficient.

**Comment by borkdude** [comment]
Can you explain why `:min-bb-version` in `bb.edn` isn't sufficient?

**Comment by waffletower-reify** [comment]
For the use case where babashka is invoked as a standalone script (often preferred) rather than using a bb.edn.

**Comment by bobisageek** [comment]
Just spitballing... for the use case of a standalone script, would it maybe make sense to expose something like the `min-bb-version` functionality from within a script more generally? I'm thinking if it were added to `add-deps`, that might lead to a desire to add it to adding classpath, loading pods, and maybe eventually to use for general bb functionality. A more general thing might also allow for deciding whether to warn or throw or quit or whatever without creating more options on `add-deps`.

I think with bb as it is, we could use `(System/getProperty "babashka.version")` to get the version and hand-roll the version check, but that's probably a bit meh.

**Comment by borkdude** [comment]
yeah, it would be nice to have something like `(babashka.whatever/assert-min-bb-version "1.12.153")` or so
